### PR TITLE
Improve comparability

### LINKED ISSUE #1: Add more scenarios
We introduced benchmarking on FastRoute some time ago (https://github.com/nikic/FastRoute/blob/master/benchmark/RealLifeExample.php) and have some scenarios which are missing here: invalid route and method not allowed.

I did observe that FastRoute was faster than SF for happy paths but not for `NOT FOUND` or `METHOD NOT ALLOWED`, maybe we get similar results for hacky-routing.

### PR DESCRIPTION
**Pull request by lcobucci**
I'm working on a new strategy for FastRoute and was using the benchmark to see how good things were. After spending some time on it, I've discovered some minor things that should be improved to allow us to better compare results.

This also fixes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,87 +118,85 @@ https://github.com/azjezz/benchmark-php-routing/actions
...
-| fast-route(mark):cached(file)        | benchLongest | 300    | 0.172731 seconds | 1736.8054041072 |
-| fast-route(group_count):cached(file) | benchAll     | 364    | 0.209652 seconds | 1736.2109208683 |
-| fast-route(group_pos):cached(file)   | benchLongest | 300    | 0.177259 seconds | 1692.4390499786 |
-| fast-route(group_count):cached(file) | benchLongest | 300    | 0.178499 seconds | 1680.6818409492 |
-| fast-route(char_count):cached(file)  | benchLongest | 300    | 0.182353 seconds | 1645.1605817657 |
-| symfony:cached(file)                 | benchAll     | 364    | 0.235754 seconds | 1543.9837625136 |
-| symfony:cached(file)                 | benchLast    | 300    | 0.197746 seconds | 1517.0973960914 |
-| symfony:cached(file)                 | benchLongest | 300    | 0.215139 seconds | 1394.447879888  |
-+--------------------------------------+--------------+--------+------------------+-----------------+
++------------------------------ Benchmark results for group cached. -------------+-----------------+
```
Comment (azjezz) [inline README.md:121]: can you update with the result from github actions? this looks quite different.

Comment (lcobucci) [reply README.md:121]: I surely can the only issue is that the result from GHA isn't reliable atm. Opcache optimisations don't kick in if the cache files aren't generated on the same process.

I can add a commit to have a dummy run (quick and dirty solution) before the real one.

Comment (azjezz) [reply README.md:121]: yep, let's do that

Comment (azjezz) [reply README.md:121]: maybe a new flag like --warmup

Comment (lcobucci) [reply README.md:121]: Oops didn't see your comment before pushing, sorry.

https://github.com/azjezz/benchmark-php-routing/pull/4/checks?check_run_id=3003594776 now shows opcache as the fastest option (vs https://github.com/azjezz/benchmark-php-routing/pull/4/checks?check_run_id=3003413955).

I'll update the README.

Would you prefer to have the script changed to use that `--warmup` flag?

Comment (lcobucci) [reply README.md:121]: README updated.
The results are far different from what I get locally, though...

File: README.md [hunk 2]
```diff
@@ -124,87 +118,85 @@ https://github.com/azjezz/benchmark-php-routing/actions
...
+| Case                          | Scenario           | Routes | Time             | Per Second      |
++-------------------------------+--------------------+--------+------------------+-----------------+
+| fast-route(mark):cached(file) | benchAll           | 364    | 0.004870 seconds | 74744.279643592 |
```
Comment (lcobucci) [inline README.md:124]: In my laptop SF router handles this in 0.002404 seconds vs 0.003114 seconds of FastRoute

Comment (azjezz) [reply README.md:124]: i would consider this to be more consistent, but we can investigate that later, i will do another run locally tomorrow and see

**Pull request merged**